react-app/src/containers/EditScreenContainer.js

```diff
@@ -108,10 +110,11 @@ class EditScreenContainer extends React.PureComponent {
           secondaryButton={avoidDiscardChangesBtn}
         />
         <EditScreen
-          date={this._initialDate}
-          text={this._initialText}
+          date={date}
+          text={text}
           onBack={this._onBack}
           onSave={this._onSave}
+          onDateChange={this._onDateChange}
```

Comment:
Shouldn't we use an intermediate function that calls `_toggleDiscardCangesModalVisibility` first?
So a date change doesn't discard the edits.